server/kafka/consumer_test.go

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019-2022 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package kafka
+
+import (
+	"encoding/binary"
+	"testing"
+
+	"github.com/linkedin/goavro/v2"
+	"github.com/riferrei/srclient"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeserializePayload(t *testing.T) {
+	server := NewMockSchemaServer(t)
+	defer server.Close()
+
+	consumer := saramaConsumer{
+		schemaRegistryOn:     true,
+		schemaRegistryClient: srclient.CreateSchemaRegistryClient(server.GetServerURL()),
+	}
+
+	avroCodec, _ := goavro.NewCodec(avroSchema)
+	native, _, _ := avroCodec.NativeFromTextual([]byte(avroMessage))
+	avroBytes, _ := avroCodec.BinaryFromNative(nil, native)
+
+	schemaIDBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(schemaIDBytes, uint32(avroSchemaID))
+
+	var payload []byte
+	payload = append(payload, byte(0))
+	payload = append(payload, schemaIDBytes...)
+	payload = append(payload, avroBytes...)
+
+	_, avroError := (*saramaConsumer).deserializePayload(&consumer, payload)
```

Comment:
I think we can simplify this.
```suggestion
	_, avroError := consumer.deserializePayload(payload)
```

Reply:
Done.